sprokit/processes/adapters/embedded_pipeline.cxx

```diff
@@ -131,19 +134,36 @@ ::embedded_pipeline()
   m_priv->m_context = std::make_shared< real_context >( m_priv );
 }
 
+// ----------------------------------------------------------------------------
 embedded_pipeline
 ::~embedded_pipeline()
 {
 }
 
-// ------------------------------------------------------------------
+// ----------------------------------------------------------------------------
+void
+embedded_pipeline
+::set_application_information( std::string const& app_name,
+                               std::string const& app_version,
+                               std::string const& app_prefix )
+{
+  m_priv->m_app_name = app_name;
+  m_priv->m_app_version = app_version;
+  m_priv->m_app_prefix = app_prefix;
+}
+
+// ----------------------------------------------------------------------------
 void
 embedded_pipeline
 ::build_pipeline( std::istream& istr, std::string const& def_dir )
 {
   // create a pipeline
   sprokit::pipeline_builder builder;
 
+  builder.add_search_path(
+    vital::application_config_file_paths(
+      m_priv->m_app_name, m_priv->m_app_version, m_priv->m_app_prefix ) );
+
```

Comment:
Will this work if these strings are not empty? There is no guarantee that `set_application_information` will be called.

Reply:
Did you mean 'if they *are* empty'? If they're empty, we simply won't use application-specific search paths. That's the intended behavior. (Note that's also what d682bb55adf509f9eed998d0f7b96047ecc7a6c9 does...)